cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/lib/partitioned/ConsumerWorkingSet.java

```diff
@@ -0,0 +1,171 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.data2.dataset2.lib.partitioned;
+
+import co.cask.cdap.api.common.Bytes;
+import co.cask.cdap.api.dataset.lib.PartitionConsumerState;
+import co.cask.cdap.api.dataset.lib.PartitionKey;
+import co.cask.cdap.proto.codec.PartitionKeyCodec;
+import com.google.common.primitives.Ints;
+import com.google.common.primitives.Longs;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+
+import java.nio.ByteBuffer;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Keeps track of a list of partitions that are either available for consuming or are currently being consumed.
+ */
+public class ConsumerWorkingSet {
+
+  private static final Gson GSON =
+    new GsonBuilder().registerTypeAdapter(PartitionKey.class, new PartitionKeyCodec()).create();
+
+  private final List<ConsumablePartition> partitions;
+
+  private PartitionConsumerState partitionConsumerState;
+
+  /**
+   * Constructs an empty working set.
+   */
+  public ConsumerWorkingSet() {
+    this(PartitionConsumerState.FROM_BEGINNING,
+         new ArrayList<ConsumablePartition>());
+  }
+
+  /**
+   * Constructs a working set using the given PartitionConsumerState and list of ConsumablePartitions.
+   */
+  private ConsumerWorkingSet(PartitionConsumerState partitionConsumerState,
+                             List<ConsumablePartition> partitions) {
+    this.partitionConsumerState = partitionConsumerState;
+    this.partitions = partitions;
+  }
+
+  public List<ConsumablePartition> getPartitions() {
```

Comment:
made the methods package-local, but still added some comments around the methods